__fixtures__/test-project/api/src/lib/auth.ts

```diff
@@ -53,24 +53,34 @@ export const hasRole = ({ roles }: { roles: AllowedRoles }): boolean => {
     return false
   }
 
-  if (roles) {
-    if (Array.isArray(roles)) {
-      if (Array.isArray(context.currentUser.roles)) {
-        return context.currentUser.roles?.some((r) => roles.includes(r))
-      } else {
-        roles.some((r) => context.currentUser.roles?.includes(r))
-      }
-    }
+  const currentUserRoles = context.currentUser?.roles as string | string[]
 
-    if (typeof roles === 'string') {
-      return context.currentUser.roles?.includes(roles)
+  if (typeof roles === 'string') {
+    if (typeof currentUserRoles === 'string') {
+      // roleToCheck is a string, currentUser.roles is a string
```

Comment:
The variable is named `roles`, but the comment says `roleToCheck`. Should we align those?

Reply:
Intentionally done, don't want to change directives, etc. and the destructure syntax to rename looks bad

Reply:
Also I don't think I should change the comment to match the variable name. It makes it cleaarer what the purpose of roles is.

Reply:
I don't agree. To me this looks confusing. Like "What's `roleToCheck`? I don't see anything with that name."
How about a comment explaining what `roles` is, and then keep using `roles` everywhere?

Like

```
// `roles` is the role, or roles, that the user needs to have at least one of
// for this check to be successful. So we check for these roles against
// `currentUser.roles`
```

Maybe that turned out too wordy, but you get the idea...

Reply:
Ok I found a nice compromise. Watcha think? 

I really dislike long wordy comments in this file, especially not next to the code... we just have so much info on there!

Reply:
> I really dislike long wordy comments in this file, especially not next to the code... we just have so much info on there!

Agreed! But maybe it really is mostly helpful for people the first time they see it? And they might not generate new RW projects as often? I mean, I'm just not sure if I'm more annoyed by it because I generate new RW apps almost daily to test some thing 😅  But this is probably a discussion for some other time.

I'll have a look at the compromise now!

Reply:
I like it!